Quaternion/tests/test_all.py

```diff
@@ -134,17 +134,25 @@ def test_from_eq_vectorized():
 
 def test_from_eq_shapes():
     q = Quat(equatorial=equatorial_23[0, 0])
```

Comment:
I *think* that this is the same as `Quat(equatorial=[10, 20, 30])`.  So just write it like that so it is dead simple to look at the single line of code and know what it is supposed to be doing.

From here, we apparently have an effective API requirement that every attribute for a scalar quaternion is a pure Python float.  (Did I get that right?)  The `.shape` test is not strict enough since `np.array(0.0)` meets that.  So all these tests should be like `type(q.ra0) is float`.

I just went onto this branch and see:
```
In[3]: q = Quat([10, 20, 30])

In [4]: type(q.ra)
Out[4]: float

In [5]: type(q.ra0)
Out[5]: numpy.ndarray
```

Having said that, it seems that our current release 3.4.1 is not consistent either:
```
In [1]: from Quaternion import Quat

In [2]: q = Quat([10, 20, 30])

In [3]: type(q.ra0)
Out[3]: numpy.float64

In [4]: type(q.ra)
Out[4]: numpy.int64  # YCK!!
```

Reply:
type(np.array(0.0)) is not  float but works, so checking "is float" is not right.

Your last comment happens because the input is integer. To ensure consistency, we can set the dtype to float in the constructor

Reply:
one could also use np.isscalar

Reply:
Maybe I was not clear enough. What I meant is that getting np.array(0.0) is perfectly fine with me. I see no reason to forbid that output.

Reply:
@taldcroft I'm not clear.  Are your requested actions to:

   * change to `Quat(equatorial=[10, 20, 30])` in `test_from_eq_shapes`?
   * verify the types as well as the shapes for all the return types (with method to verify type tbd)

?

I would push back that both seem a bit out of scope for this PR, but we're obviously here because the testing isn't perfect and our API has been largely implicit.